src/Microsoft.Health.Fhir.SqlServer/Features/Operations/Import/ImportOrchestratorJob.cs

```diff
@@ -109,34 +105,11 @@ public async Task<string> ExecuteAsync(JobInfo jobInfo, IProgress<string> progre
             {
                 cancellationToken.ThrowIfCancellationRequested();
 
-                if (currentResult.Progress == ImportOrchestratorJobProgress.Initialized)
-                {
-                    await ValidateResourcesAsync(inputData, cancellationToken);
-
-                    currentResult.Progress = ImportOrchestratorJobProgress.InputResourcesValidated;
-                    progress.Report(JsonConvert.SerializeObject(currentResult));
-
-                    _logger.LogJobInformation(jobInfo, "Input Resources Validated.");
-                }
-
-                if (currentResult.Progress == ImportOrchestratorJobProgress.InputResourcesValidated)
-                {
-                    await _importOrchestratorJobDataStoreOperation.PreprocessAsync(cancellationToken);
-
-                    currentResult.Progress = ImportOrchestratorJobProgress.PreprocessCompleted;
-                    progress.Report(JsonConvert.SerializeObject(currentResult));
+                await ValidateResourcesAsync(inputData, cancellationToken);
+                _logger.LogJobInformation(jobInfo, "Input Resources Validated.");
 
-                    _logger.LogJobInformation(jobInfo, "Preprocess Completed.");
-                }
-
-                if (currentResult.Progress == ImportOrchestratorJobProgress.PreprocessCompleted)
-                {
-                    await ExecuteImportProcessingJobAsync(progress, jobInfo, inputData, currentResult, cancellationToken);
-                    currentResult.Progress = ImportOrchestratorJobProgress.SubJobsCompleted;
-                    progress.Report(JsonConvert.SerializeObject(currentResult));
-
-                    _logger.LogJobInformation(jobInfo, "SubJobs Completed.");
-                }
+                await ExecuteImportProcessingJobAsync(jobInfo, inputData, currentResult, cancellationToken);
```

Comment:
Since the orchestrator is long running, what happens if the job gets restarted (either through a crash, deployment, etc)? Will the import start again and queue up duplicate child jobs?

Reply:
Import orchestrator will be restarted. It will try enqueue the same jobs, but because jobs definitions are identical to previous ones, jobs will not be enqueued, but old ones will be returned back.

Reply:
Sick - that's awesome. So if our orchestrators produce the same result, we don't need all this progress work. One minor suggestion would be to add a comment for those unfamiliar with the code, but not required at all.

Reply:
If add, then it is per each queue type. Too many places...